linkup-cli/src/services/local_server.rs

```diff
@@ -0,0 +1,174 @@
+use std::{
+    fs::File,
+    os::unix::process::CommandExt,
+    path::PathBuf,
+    process::{self, Stdio},
+    time::Duration,
+};
+
+use reqwest::StatusCode;
+use tokio::time::sleep;
+use url::Url;
+
+use crate::{
+    linkup_file_path,
+    local_config::{upload_state, LocalState},
+    signal, worker_client,
+};
+
+use super::BackgroundService;
+
+pub const LINKUP_LOCAL_SERVER_PORT: u16 = 9066;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Failed while handing file: {0}")]
+    FileHandling(#[from] std::io::Error),
+    #[error("Failed to stop pid: {0}")]
+    StoppingPid(#[from] signal::PidError),
+    #[error("Failed to reach the local server")]
+    ServerUnreachable,
+    #[error("WorkerClient error: {0}")]
+    WorkerClient(#[from] worker_client::Error),
+}
+
+pub struct LocalServer {
+    stdout_file_path: PathBuf,
+    stderr_file_path: PathBuf,
+    pidfile_path: PathBuf,
+}
+
+impl LocalServer {
+    pub fn new() -> Self {
+        Self {
+            stdout_file_path: linkup_file_path("localserver-stdout"),
+            stderr_file_path: linkup_file_path("localserver-stderr"),
+            pidfile_path: linkup_file_path("localserver-pid"),
+        }
+    }
+
+    pub fn url() -> Url {
+        Url::parse(&format!("http://localhost:{}", LINKUP_LOCAL_SERVER_PORT))
+            .expect("linkup url invalid")
+    }
+
+    fn start(&self) -> Result<(), Error> {
+        log::debug!("Starting {}", Self::NAME);
+
+        let stdout_file = File::create(&self.stdout_file_path)?;
+        let stderr_file = File::create(&self.stderr_file_path)?;
+
+        process::Command::new("linkup")
```

Comment:
How does this work when testing with `cargo start` - this will always call the "old compiled code"?

Reply:
That is a good point! Fixed it in 88e13afc533feb26fe294f8f5b96aa79b8cc4f04